pkg/secrethub/client.go

```diff
@@ -144,7 +144,7 @@ func NewClient(with ...ClientOption) (*Client, error) {
 		var provider credentials.Provider
 		switch strings.ToLower(identityProvider) {
 		case "", "key":
-			provider = credentials.UseKey(client.DefaultCredential())
+			provider = credentials.UseKey(client.DefaultCredential(), credentials.DefaultKeyDecoder())
```

Comment:
Good point! This will significantly simplify the code.

Reply:
An alternative decoder is the one which prompts for a passphrase. It is used in the CLI if the credential is password protected.

But you're right, this could be simplified. Maybe we can make the default `credential.UseKey` just use the default key decoder and have a separate function if you want to provide a password. This way we can keep the passwordless keys simple to use and still have an alternative for password protected credentials.

What do you think?